src/deploy/functions/params.ts

```diff
@@ -326,12 +326,37 @@ export async function resolveParams(
         "Parameter " + param.name + " has default value " + paramDefault + " of wrong type"
       );
     }
-    paramValues[param.name] = await promptParam(param, projectId, paramDefault);
+    paramValues[param.name] = await promptParam(param, firebaseConfig.projectId, paramDefault);
   }
 
   return paramValues;
 }
 
+function populateDefaultParams(config: FirebaseConfig): Record<string, ParamValue> {
+  const defaultParams: Record<string, ParamValue> = {};
+  defaultParams["DATABASE_URL"] = new ParamValue(config.databaseURL, true, {
+    string: true,
+    boolean: false,
+    number: false,
+  });
+  defaultParams["PROJECT_ID"] = new ParamValue(config.projectId, true, {
+    string: true,
+    boolean: false,
+    number: false,
+  });
+  defaultParams["GCLOUD_PROJECT"] = new ParamValue(config.projectId, true, {
+    string: true,
+    boolean: false,
+    number: false,
+  });
+  defaultParams["STORAGE_BUCKET"] = new ParamValue(config.storageBucket, true, {
+    string: true,
+    boolean: false,
+    number: false,
```

Comment:
I'm pretty sure our type for FirebaseConfig is wrong. `databaseURL` and `storageBucket` may be empty for a project that didn't initialize neither of these products. Should we instead conditionally setup those parameters?

Reply:
I think so, yeah.